File: ZenCare.Mobile/lib/models/reset_password_request.dart
class ResetPasswordRequest {
  ResetPasswordRequest({
    required this.token,
    required this.newPassword,
    required this.confirmNewPassword,
  });

  final String token;
  final String newPassword;
  final String confirmNewPassword;

  Map<String, dynamic> toJson() {
    return {
      'token': token,
      'newPassword': newPassword,
      'confirmNewPassword': confirmNewPassword,
    };
  }
}

File: ZenCare.Model/Requests/ResetPasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace ZenCare.Model.Requests;

public class ResetPasswordRequest
{
    [Required(ErrorMessage = "Reset token is required.")]
    public string Token { get; set; } = string.Empty;

    [Required(ErrorMessage = "New password is required.")]
    [MinLength(6, ErrorMessage = "New password must contain at least 6 characters.")]
    public string NewPassword { get; set; } = string.Empty;

    [Required(ErrorMessage = "Confirm new password is required.")]
    [Compare(nameof(NewPassword), ErrorMessage = "Passwords do not match.")]
    public string ConfirmNewPassword { get; set; } = string.Empty;
}
